SandWorm/Analytics/Slope.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Threading.Tasks;
+using Rhino.Display;
+using Rhino.Geometry;
+
+namespace SandWorm.Analytics
+{
+    public class Slope : Analysis.MeshColorAnalysis
+    {
+        readonly int maximumSlope = 1000; // Needs to be high to account for stray values
+
+        public Slope() : base("Visualise Slope")
+        {
+        }
+
+        public void GetColorCloudForAnalysis(ref Color[] vertexColors, double[] pixelArray, int x, int y, double deltaX, double deltaY)
+        {
+            if (lookupTable == null)
+            {
+                ComputeLookupTableForAnalysis(0.0);
+            }
+
+            // Get slope values for given array of pixels
+            var slopeValues = CalculateSlope(pixelArray, x, y, deltaX, deltaY);
+
+            // Lookup slope value in color table
+            vertexColors = new Color[pixelArray.Length];
+            for (int i = 0; i < pixelArray.Length; i++)
+            {
+                var slopeValue = slopeValues[i];
+                var slopeColor = lookupTable[slopeValue];
+                vertexColors[i] = slopeColor; 
+            }
+        }
+
+        public override void ComputeLookupTableForAnalysis(double sensorElevation)
+        {
+            var slightSlopeRange = new Analysis.VisualisationRangeWithColor
+            {
+                ValueSpan = 30,
+                ColorStart = new ColorHSL(0.30, 1.0, 0.5), // Green
+                ColorEnd = new ColorHSL(0.15, 1.0, 0.5) // Yellow
+            };
+            var moderateSlopeRange = new Analysis.VisualisationRangeWithColor
+            {
+                ValueSpan = 30,
+                ColorStart = new ColorHSL(0.15, 1.0, 0.5), // Green
+                ColorEnd = new ColorHSL(0.0, 1.0, 0.5) // Red
+            };
+            var extremeSlopeRange = new Analysis.VisualisationRangeWithColor
+            {
+                ValueSpan = maximumSlope - slightSlopeRange.ValueSpan - moderateSlopeRange.ValueSpan,
+                ColorStart = new ColorHSL(0.0, 1.0, 0.5), // Red
+                ColorEnd = new ColorHSL(0.0, 1.0, 0.0) // Black
+            };
+            ComputeLinearRanges(slightSlopeRange, moderateSlopeRange, extremeSlopeRange);
+        }
+        
+        public ushort[] CalculateSlope(double[] pixelArray, int width, int height, double deltaX, double deltaY)
```

Comment:
There were some minor typos in the comments to this function, which I fixed last night. Could you please copy these over from the updated branch?